libfreerdp/core/input.c

```diff
@@ -903,6 +904,14 @@ BOOL freerdp_input_send_keyboard_event(rdpInput* input, UINT16 flags, UINT8 code
 	if (freerdp_settings_get_bool(input->context->settings, FreeRDP_SuspendInput))
 		return TRUE;
 
+	/* Save last input timestamp in prevent-session-lock mode */
+	if (input->context->settings->FakeMouseMotionInterval)
```

Comment:
best move this to a `static` function so it can easily be called for each location we require it

```
static BOOL input_update_last_event(rdpInput* input, BOOL mouse, UINT16  x, UINT16 y) {
  rdp_input_internal* int = input_cast(input);
  WINPR_ASSERT(input);
  WINPR_ASSERT(input->context);
  if (freerdp_settings_get_uint32(input->context->settings, FreeRDP_FakeMouseMotionInterval) > 0)
  {
		struct timespec ts={0};
		if (timespec_get(&ts, TIME_UTC) == 0)
                  return FALSE;
		int->lastInputTimestamp = ts.tv_sec;
                if (mouse) {
                  int->lastX = x;
                  int->lastY = y;
               }
  }
  return TRUE;
}
```